Ginger/GingerCoreNET/ActionsLib/APIModelActionConversion/ApiActionConversionUtils.cs

```diff
@@ -0,0 +1,461 @@
+                            for (int activityIndex = 0; activityIndex < bf.BusinessFlow.Activities.Count(); activityIndex++)
+                            {
+                                Activity activity = bf.BusinessFlow.Activities[activityIndex];
+                                RepositoryItemKey ta = WorkSpace.Instance.Solution.ApplicationPlatforms.Where(x => x.ItemName == activity.TargetApplication).FirstOrDefault().Key;
+                                if (activity != null && activity.Active && WorkSpace.Instance.Solution.GetApplicationPlatformForTargetApp(ta.ItemName) == ePlatformType.WebServices)
+                                {
+                                    ConvertAction(parameterizeRequestBody, pullValidations, activity, ta);
+                                }
+                            }
+                        }
+                        bf.ConversionStatus = eConversionStatus.Finish;
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Reporter.ToLog(eLogLevel.ERROR, "Error occurred while trying to convert action", ex);
+                }
+            }
+        }
+
+        /// <summary>
+        /// This method will convert the action
+        /// </summary>
+        /// <param name="parameterizeRequestBody"></param>
+        /// <param name="pullValidations"></param>
+        /// <param name="activity"></param>
+        /// <param name="ta"></param>
+        private void ConvertAction(bool parameterizeRequestBody, bool pullValidations, Activity activity, RepositoryItemKey ta)
+        {
+            for (int actionIndex = 0; actionIndex < activity.Acts.Count; actionIndex++)
+            {
+                try
+                {
+                    Act act = (Act)activity.Acts[actionIndex];
+                    if (act.Active && (act.GetType() == typeof(ActWebAPIRest) || act.GetType() == typeof(ActWebAPISoap)))
+                    {
+                        // get the index of the action that is being converted 
+                        int selectedActIndex = activity.Acts.IndexOf(act); 
+
+                        //Create/Update API Model
+                        bool isModelExists = true;
+                        ApplicationAPIModel applicationModel = GetAPIModelIfExists(act);
+
+                        if (applicationModel == null)
+                        {
+                            isModelExists = false;
+                            applicationModel = new ApplicationAPIModel();
+                            applicationModel.TargetApplicationKey = ta;
+                            SetApplicationAPIModel(ref applicationModel, act, pullValidations);
+                        }
+
+                        //Create WebAPIModel action
+                        ActWebAPIModel actApiModel = GetNewAPIModelAction(applicationModel.Guid, act);
+                        AddAppParameters(applicationModel, act, actApiModel, parameterizeRequestBody);
```

Comment:
calling AddAppParameters here means you will do it no matter if the API model created new or if it already exist...
we do not want to parameterize it if API model already exist- do the parameterize only for new API models.
learning optional value should be done for both cases, new or existing API model